### ISSUE TITLE
Stabilize the ConfirmDialoc component

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60027]
### Description

See https://github.com/WordPress/gutenberg/pull/59825#issuecomment-2006613187

The `ConfirmDialog` component was merged more than 2 years ago in https://github.com/WordPress/gutenberg/pull/34153 but it is still marked as 'experimental'. Worth considering to stabilize it.

To recap why it was introduced, see the related issues https://github.com/WordPress/gutenberg/issues/16583 and  https://github.com/WordPress/gutenberg/issues/33937

- The native `window.confirm()` isn't an ideal experience as we're unable to customise the OK button to have a more descriptive label.
- Major browsers - starting with Chrome - will soon restrict the alert/confirm/prompt functions from being used from within cross-origin iframes.

The `ConfirmDialog` component seems to work well tor the editor needs. From an accessibility perspective, I think it only needs a lable for the dialot (the element with role=dialog) as it's currently unlabeled.

### Step-by-step reproduction instructions

N/A
To see an example of a `ConfirmDialog` in action:
- Go to the site editor, view mode.
- In the navigation panel, click Design > Navigation > {menu name} > Actions ellipsis button > Delete
- Observe the confirmation dialog.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by manzoorwanijk** [comment]
Any update on this? The component seems to be pretty stable.

**Comment by talldan** [comment]
This is already tracked by https://github.com/WordPress/gutenberg/issues/59418. There's a PR open - https://github.com/WordPress/gutenberg/pull/61014.
